openstack/compute/v2/extensions/hypervisors/requests.go

```diff
@@ -5,9 +5,48 @@ import (
 	"github.com/gophercloud/gophercloud/pagination"
 )
 
+// ListOptsBuilder allows extensions to add additional parameters to the
+// List request.
+type ListOptsBuilder interface {
+	ToHypervisorListQuery() (string, error)
+}
+
+// ListOpts allows the filtering and sorting of paginated collections through
+// the API. Filtering is achieved by passing in struct field values that map to
+// the server attributes you want to see returned. Marker and Limit are used
+// for pagination.
+type ListOpts struct {
```

Comment:
All 4 of the options in this struct require a specific microversion to be set. Please see the "New Request Fields" section of our [microversions](https://github.com/gophercloud/gophercloud/blob/master/docs/MICROVERSIONS.md#new-request-fields) document for how to handle this.

Basically, please add a doc comment stating which microversion is required and change the type of each parameter to a pointer (`*string`, `*int`, etc). For example:

```go
// Marker is a ID of the last-seen item as a UUID.
// This requires microversion 2.33
Marker *string `q:"marker"`
```

(note: `q` tags don't support `omitempty`, so you don't need to add them here)

Reply:
I've checked that `Limit` and `Marker` fields don't need to be a pointer:

https://github.com/gophercloud/gophercloud/blob/fa957f3e1f14997f31f301788af11e247d1facc6/openstack/loadbalancer/v2/listeners/requests.go#L44

I changed the other fields to pointer and kept these two.

Reply:
`Limit` and `Marker` are part of the standard API set for the LoadBalancer service. However, they were added via a microversion for the Compute Hypervisor API. You can see this in the following code, where there are two blocks of code that define these two parameters for microversions 2.33 to 2.52 and then 2.53 and up:

https://github.com/openstack/nova/blob/stable/victoria/nova/api/openstack/compute/hypervisors.py#L196-L215

It's also in the API documentation: https://docs.openstack.org/api-ref/compute/?expanded=list-hypervisors-details-detail#hypervisors-os-hypervisors

Please change these to pointers.